syntax = "proto3";

package transaction_verification;

service TransactionVerificationService {
    rpc verifyTransaction (TransactionRequest) returns (TransactionResponse);
}

message Item {
    string name = 1;
    int32 quantity = 2;
}

message CreditCard {
    string number = 1;
    string expiration_date = 2;
    string cvv = 3;
}

message TransactionRequest {
    string user_name = 1;
    string user_contact = 2;
    repeated Item items = 3;
    CreditCard credit_card = 4;
    bool terms_accepted = 5;
}

message TransactionResponse {
    bool is_valid = 1;
    string reason = 2;
}
